configure.ac

```diff
@@ -526,6 +526,60 @@ if test "x$enable_unwind" = xno; then
    AC_SEARCH_LIBS([backtrace], [execinfo])
 fi
 
+AC_ARG_ENABLE([backtraces],
+              [AS_HELP_STRING([--enable-backtraces=<unwind-ptrace>],
+              [enable printing backtraces.])],
+              [],
+              [enable_backtraces=no])
+case "$enable_backtraces" in
+   no)
+      ;;
+   unwind-ptrace)
+      AC_CHECK_LIB([unwind-ptrace], [_UPT_create], [], [AC_MSG_ERROR([can not find required library libunwind-ptrace])], [-lunwind-generic])
+      AC_CHECK_LIB([unwind-generic], [_UPT_create], [], [AC_MSG_ERROR([can not find required library libunwind-generic])])
+      AC_CHECK_HEADERS([libunwind-ptrace.h], [], [
+         CFLAGS="$CFLAGS -I/usr/include/libunwind"
+         AC_CHECK_HEADERS([libunwind/libunwind-ptrace.h], [], [AC_MSG_ERROR([can not find required header file libunwind-ptrace.h])])
+      ])
+      ;;
+   *)
+      AC_MSG_ERROR([bad value '$enable_backtraces' for --enable-backtraces])
+      ;;
+esac
+if test "x$enable_backtraces" = xunwind-ptrace; then
+   AC_DEFINE([BACKTRACE_ENABLED], [1], [Define if the backtraces is enabled])
```

Comment:
How about just name it `ENABLE_BACKTRACE`?
Same wording for all `--enable-*` configure options in Autoconf.

Reply:
And by the way, I don't like the idea of using plural for the "`--enable-backtraces`" option. The term "backtrace" may be used as an adjective or a verb, and the singular/infinitive form is more widely seen in web searches.

Reply:
Any objections to `--enable-backtrace-support` then? I know it's kinda long, but it avoids the issue with postfixing the `enable` …

Reply:
Let me state my opinion.
1. For macro name, `BACKTRACE_SUPPORT` is fine, even though it's not the first choice if I were able to vote.
2. For configure option, I would say `--enable-backtrace` is my first preference and `--enable-backtrace-support` second. For a user facing option it's better to name it concisely and not overly verbose.

Reply:
Let's wait for some more voices. @cgzones @fasterit @natoscott ?

Reply:
I think `--enable-backtrace` is best as we don't do `--enable-unwind-support` etc. either

The macro names seem to be commonly `HAVE_*`. I would let @BenBE do the final call here as he is typically closest to the coding style.

Greets
/DLange

Reply:
Following up internally, we'll go with:
- `--enable-backtrace`
- `HAVE_BACKTRACE`

Though I'm not fully happy with this, this is, based on discussions, the most consistent way to go forward in this case IMO. It also matches best with what we already use for other feature flags in our code.

Reply:
@MrCirdo Implement both. Let `--enable-backtrace` translate to `--enable-backtrace=unwind-ptrace` in Linux, and `--enable-backtrace=no` for other platforms. In case you don't understand how to use Autoconf macros, feel free to ask and I can offer help when I can.

Reply:
Oh okay, I see. I'll try.